x/lazyledgerapp/keeper/msg_server.go

```diff
@@ -1,15 +1,27 @@
 package keeper
 
-import "github.com/lazyledger/lazyledger-app/x/lazyledgerapp/types"
+import (
+	"context"
+
+	"github.com/lazyledger/lazyledger-app/x/lazyledgerapp/types"
+)
 
 var _ types.MsgServer = msgServer{}
 
+// MsgServer is the server API for Msg service.
+type MsgServer interface {
+	// PayForMessage allows the user to post data to made be available.
+	PayForMessage(context.Context, *types.MsgWirePayForMessage) (*types.MsgPayForMessageResponse, error)
+	// PayForMessage allows the user to post data to made be available.
+	SignedTransactionDataPayForMessage(context.Context, *types.SignedTransactionDataPayForMessage) (*types.SignedTransactionDataPayForMessageResponse, error)
+}
+
```

Comment:
What is the relation of this interface to the `MsgServer interface` in `x/lazyledgerapp/types/tx.pb.go`?

Reply:
The `MsgServer` interface in `x/lazyledgerapp/types/tx.pb.go` only accounts for rpc endpoints, as the generated code assumes that there will always be 1 enpoint per 1 handler. That is not the case in our situation, where we need two handlers, but only 1 endpoint.

Reply:
What was the reason again to only have one endpoint here?

Reply:
because each `MsgWirePayForMessage` will have multiple `SignedTransactionDataPayForMessage`s in it, and one `SignedTransactionDataPayForMessage` is picked during `PreprocessTxs`, so having an endpoint for `SignedTransactionDataPayForMessage` could be confusing.

Reply:
That makes sense!